osu.Game/Rulesets/Replays/FramedReplayInputHandler.cs

```diff
@@ -164,12 +162,15 @@ private void updateDirection(double time)
             if (!CurrentTime.HasValue)
             {
                 currentDirection = 1;
+                return;
             }
-            else
-            {
-                currentDirection = time.CompareTo(CurrentTime);
-                if (currentDirection == 0) currentDirection = 1;
-            }
+
+            var newDirection = CurrentTime <= time ? 1 : -1;
+
+            if (currentDirection == newDirection) return;
+
+            currentFrameIndex = clampedNextFrameIndex;
```

Comment:
So as far as I can tell, this current frame index change is the only actual functional change in this PR, correct? Can you write a test case that you would expect to fail without this and maybe we can proceed from there?

This replay stuff is super fragile and touching it without adding tests is just asking for trouble further down the line.

Reply:
I agree and the PR was rushed and I didn't even understand the original code.
I'll probably PR more substantial rewrite in the next year.